consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -386,21 +386,21 @@ where
         let mut new: Self = self.clone();
         new.branches = new.branches.apply_header(id, parent, slot)?;
         new.local_chain = new.fork_choice();
-        let pruned_blocks = new.update_lib().collect();
+        let pruned_blocks = new.update_lib();
         Ok(UpdatedCryptarchia {
             cryptarchia: new,
             pruned_blocks,
         })
     }
 
-    fn update_lib(&mut self) -> impl Iterator<Item = Id> + '_ {
+    fn update_lib(&mut self) -> HashSet<Id> {
```

Comment:
maybe a comment on what is returned by this function, it's not immediately obvious from the signature